jax/interpreters/pxla.py

```diff
@@ -590,7 +590,7 @@ def _hashable_index(idx):
 # TODO(skye): is there a simpler way to rewrite this using sharding_spec?
 def _shard_sharded_device_array_slow_path(x, devices, indices):
   candidates = defaultdict(list)
-  for buf, idx in zip(x.device_buffers, x.indices):
+  for buf, idx in safe_zip(x.device_buffers, x.indices):
```

Comment:
I made these changes of `zip` -> `safe_zip` because while debugging my changes `zip` was failing silently.